rct229/rulesets/ashrae9012019/section18/section18rule2.py

```diff
@@ -0,0 +1,290 @@
+                        rmd_b,
+                        zones_and_terminal_unit_list_dict_b[hvac_id_b]["zone_list"],
+                    )
+                    if do_multi_zone_evaluation
+                    else []
+                )
+
+                # check if all the zones served by the system are on the same floor
+                if all([zone in zones_on_floor for zone in zones_served_by_system]):
+                    # check if there are any other systems of the same system type that serve zones on this floor
+                    for hvac_system_type2 in applicable_hvac_sys_ids_b:
+                        for hvac_sys2_id_b in baseline_system_types_dict_b[
+                            hvac_system_type2
+                        ]:
+                            # check if hvac_id_b and hvac_sys2_id_b are the same system
+                            if hvac_sys2_id_b != hvac_id_b:
+                                zones_served_by_system2 = (
+                                    zones_and_terminal_unit_list_dict_b[hvac_sys2_id_b][
+                                        "zone_list"
+                                    ]
+                                )
+                                if set(zones_served_by_system2).intersection(
+                                    set(zones_on_floor)
+                                ):
+                                    hvac_data_b[hvac_id_b][
+                                        "does_two_sys_exist_on_same_fl_b"
+                                    ] = False
+                                elif (
+                                    hvac_sys2_id_b
+                                    in lab_zone_hvac_systems["lab_zones_only"]
+                                    and len(lab_zone_hvac_systems["lab_zones_only"])
+                                    == 1
+                                    and lab_zone_exhaust_b > AIRFLOW_15000_CFM
+                                ):
+                                    hvac_data_b[hvac_id_b][
+                                        "does_two_sys_exist_on_same_fl_b"
+                                    ] = True
+
+                                elif (
+                                    hvac_sys2_id_b
+                                    in lab_zone_hvac_systems["lab_zones_only"]
+                                    and len(lab_zone_hvac_systems["lab_zones_only"])
+                                    == 1
+                                    and lab_zone_exhaust_b <= AIRFLOW_15000_CFM
+                                ):
+                                    hvac_data_b[hvac_id_b][
+                                        "does_two_sys_exist_on_same_fl_b"
+                                    ] = "undetermined"
+
+                                else:
+                                    hvac_data_b[hvac_id_b][
+                                        "does_two_sys_exist_on_same_fl_b"
+                                    ] = False
```

Comment:
If there are three labels, I think we should set all of them to a string (true, false and undetermined).
Also, I think this shall be an any logic for a list, but in this case, the logic is just keep overriding the flag with new booleans.

Reply:
Addressed.